File: frontend/src/types/doctor.ts
import { DailyHealthCheck, DailyHealthTrendPoint } from './healthCheck';
import { SessionSummary } from './questionnaire';
import { EyeAnalysisSession } from './eyeAnalysis';
import { RiskAssessment } from './risk';

export type NoteType = 'ROUTINE_REVIEW' | 'EMERGENCY_FOLLOW_UP' | 'DIAGNOSTIC_HYPOTHESIS' | 'DISCHARGE';

export interface DoctorRecentActivityItem {
  patient_id: string;
  patient_name: string;
  activity_type: 'HEALTH_CHECK' | 'QUESTIONNAIRE' | 'EYE_ANALYSIS' | 'RISK_ASSESSMENT';
  timestamp: string;
  description: string;
  risk_level?: string | null;
}

export interface DoctorDashboardSummary {
  total_assigned_patients: number;
  risk_distribution: {
    HIGH: number;
    MEDIUM: number;
    LOW: number;
    UNASSESSED: number;
  };
  recent_activity: DoctorRecentActivityItem[];
}

export interface AssignedPatientCard {
  patient_id: string;
  full_name: string;
  email: string;
  date_of_birth: string;
  gender: string;
  assigned_at: string;
  latest_risk_level?: 'LOW' | 'MEDIUM' | 'HIGH' | null;
  latest_risk_score?: number | null;
  latest_assessment_date?: string | null;
  latest_health_check_date?: string | null;
  latest_health_check_dizziness?: number | null;
  total_health_checks: number;
}

export interface DoctorPatientList {
  items: AssignedPatientCard[];
  total: number;
}

export interface DoctorPatientDossier {
  patient_id: string;
  full_name: string;
  email: string;
  date_of_birth: string;
  gender: string;
  medical_history?: string | null;
  emergency_contact_name?: string | null;
  emergency_contact_phone?: string | null;
  latest_health_check?: DailyHealthCheck | null;
  latest_questionnaire?: SessionSummary | null;
  latest_eye_analysis?: EyeAnalysisSession | null;
  latest_risk_assessment?: RiskAssessment | null;
  recent_notes_count: number;
}

export interface DoctorNote {
  id: string;
  patient_id: string;
  doctor_id: string;
  doctor_name: string;
  doctor_specialization: string;
  risk_assessment_id?: string | null;
  note_type: NoteType;
  content: string;
  is_shared_with_patient: boolean;
  created_at: string;
  updated_at: string;
}

export interface DoctorPatientReport {
  patient_id: string;
  patient_name: string;
  generated_at: string;
  health_summary: {
    total_records_14d: number;
    average_dizziness: number;
    average_imbalance: number;
    average_sleep: number;
    average_stress: number;
  };
  questionnaire_summary?: {
    session_id: string;
    completed_at?: string | null;
    answers: Array<{
      question_code: string;
      question_text: string;
      answer: any;
    }>;
  } | null;
  eye_analysis_summary?: {
    session_id: string;
    created_at: string;
    quality: any;
    features: Record<string, number>;
  } | null;
  latest_risk?: RiskAssessment | null;
  clinical_notes: DoctorNote[];
  disclaimer: string;
}

File: backend/app/schemas/doctor.py
from datetime import datetime, date
from typing import Optional, List, Dict, Any
from pydantic import BaseModel, Field, ConfigDict
from app.models.clinical import NoteType
from app.models.risk import RiskLevel
from app.schemas.monitoring import DailyHealthCheckResponse, DailyHealthTrendResponse
from app.schemas.risk import RiskAssessmentResponse
from app.schemas.questionnaire import SessionSummaryResponse
from app.schemas.eye_analysis import EyeAnalysisSessionResponse


class BaseSchema(BaseModel):
    model_config = ConfigDict(from_attributes=True)


class DoctorRecentActivityItem(BaseSchema):
    patient_id: str
    patient_name: str
    activity_type: str  # "HEALTH_CHECK", "QUESTIONNAIRE", "EYE_ANALYSIS", "RISK_ASSESSMENT"
    timestamp: datetime
    description: str
    risk_level: Optional[str] = None


class DoctorDashboardSummaryResponse(BaseSchema):
    total_assigned_patients: int
    risk_distribution: Dict[str, int]
    recent_activity: List[DoctorRecentActivityItem] = []


class AssignedPatientCardResponse(BaseSchema):
    patient_id: str
    full_name: str
    email: str
    date_of_birth: date
    gender: str
    assigned_at: datetime
    latest_risk_level: Optional[str] = None
    latest_risk_score: Optional[float] = None
    latest_assessment_date: Optional[datetime] = None
    latest_health_check_date: Optional[date] = None
    latest_health_check_dizziness: Optional[int] = None
    total_health_checks: int = 0


class DoctorPatientListResponse(BaseSchema):
    items: List[AssignedPatientCardResponse]
    total: int


class DoctorPatientDossierResponse(BaseSchema):
    patient_id: str
    full_name: str
    email: str
    date_of_birth: date
    gender: str
    medical_history: Optional[str] = None
    emergency_contact_name: Optional[str] = None
    emergency_contact_phone: Optional[str] = None
    latest_health_check: Optional[DailyHealthCheckResponse] = None
    latest_questionnaire: Optional[SessionSummaryResponse] = None
    latest_eye_analysis: Optional[EyeAnalysisSessionResponse] = None
    latest_risk_assessment: Optional[RiskAssessmentResponse] = None
    recent_notes_count: int = 0


class DoctorNoteCreateRequest(BaseSchema):
    content: str = Field(..., min_length=3, max_length=5000)
    note_type: NoteType = NoteType.ROUTINE_REVIEW
    risk_assessment_id: Optional[str] = None
    is_shared_with_patient: bool = True


class DoctorNoteUpdateRequest(BaseSchema):
    content: str = Field(..., min_length=3, max_length=5000)
    note_type: Optional[NoteType] = None
    is_shared_with_patient: Optional[bool] = None


class DoctorNoteResponse(BaseSchema):
    id: str
    patient_id: str
    doctor_id: str
    doctor_name: str
    doctor_specialization: str
    risk_assessment_id: Optional[str] = None
    note_type: NoteType
    content: str
    is_shared_with_patient: bool
    created_at: datetime
    updated_at: datetime


class DoctorPatientReportResponse(BaseSchema):
    patient_id: str
    patient_name: str
    generated_at: datetime
    health_summary: Dict[str, Any]
    questionnaire_summary: Optional[Dict[str, Any]] = None
    eye_analysis_summary: Optional[Dict[str, Any]] = None
    latest_risk: Optional[RiskAssessmentResponse] = None
    clinical_notes: List[DoctorNoteResponse] = []
    disclaimer: str = "AI-assisted screening and decision support information — not a medical diagnosis."

